operators/src/engine/query_processor.rs

```diff
@@ -82,6 +83,24 @@ where
     }
 }
 
+/// An instantiation of a plot operator that produces a stream of vector results for a query
+pub trait PlotQueryProcessor: Sync + Send {
+    type PlotType;
+
+    fn plot_query<'a>(
+        &'a self,
+        query: QueryRectangle,
+        ctx: &'a dyn QueryContext,
+    ) -> BoxFuture<'a, Result<Self::PlotType>>;
```

Comment:
is it safe to assume that there is always exactly one plot resulting from a workflow?

I could think of
- Multiple views of the same data (paginated in frontend)
 - could also be realized with a special compund plot type
- plot that is split into multiple plots for contiguous time spans
 - query a decade, get a plot for each year
 
not sure if that makes sense but we should at least think about this.

Reply:
I guess for now it is since we currently don't know what to do if there is more than one output.
We have only one HTTP reply for now, so I guess we should go with it and then have some ideas for the future.
For instance, I could think of updating plots, e.g. continuing to draw the line for new results… but it is very unclear for now.
Going from Future to Stream in the future with a good concept is possible.